In [1]:
import yfinance as yf

ticker = "AAPL"


In [2]:
data_5d = yf.download(ticker, period="5d", interval="1h")

display(data_5d.shape)

display(data_5d.head(10))

display(data_5d.tail(10))

display(data_5d.info())

display(data_5d.describe())

[*********************100%***********************]  1 of 1 completed


(26, 5)

Price                           Close        High         Low        Open  \
Ticker                           AAPL        AAPL        AAPL        AAPL   
Datetime                                                                    
2026-01-28 14:30:00+00:00  256.089996  258.855011  255.214996  257.559998   
2026-01-28 15:30:00+00:00  255.490005  256.709991  255.259995  256.059998   
2026-01-28 16:30:00+00:00  254.820007  255.479996  254.660004  255.470001   
2026-01-28 17:30:00+00:00  255.255005  255.389999  254.509995  254.800003   
2026-01-28 18:30:00+00:00  255.110001  255.880005  254.779999  255.259995   
2026-01-28 19:30:00+00:00  256.475006  256.640015  255.050003  255.080002   
2026-01-28 20:30:00+00:00  256.440002  256.899994  255.929993  256.459991   
2026-01-29 14:30:00+00:00  256.899994  259.649994  255.490005  258.070007   
2026-01-29 15:30:00+00:00  256.170013  257.040009  254.410004  256.890015   
2026-01-29 16:30:00+00:00  257.220001  257.410004  255.119995  256.130005   

Price                       Volume  
Ticker                        AAPL  
Datetime                            
2026-01-28 14:30:00+00:00  6604210  
2026-01-28 15:30:00+00:00  2745774  
2026-01-28 16:30:00+00:00  2230506  
2026-01-28 17:30:00+00:00  1683148  
2026-01-28 18:30:00+00:00  1899457  
2026-01-28 19:30:00+00:00  3391568  
2026-01-28 20:30:00+00:00  3092338  
2026-01-29 14:30:00+00:00  7282171  
2026-01-29 15:30:00+00:00  4801889  
2026-01-29 16:30:00+00:00  3524580

Price                           Close        High         Low        Open  \
Ticker                           AAPL        AAPL        AAPL        AAPL   
Datetime                                                                    
2026-02-02 18:30:00+00:00  267.179993  267.350006  266.619995  266.790009   
2026-02-02 19:30:00+00:00  268.980011  269.569885  266.820007  267.125000   
2026-02-02 20:30:00+00:00  269.959991  270.489990  268.839996  268.975006   
2026-02-03 14:30:00+00:00  269.885010  271.579987  267.690002  269.130005   
2026-02-03 15:30:00+00:00  268.894989  271.285004  267.609985  269.850006   
2026-02-03 16:30:00+00:00  268.899994  270.500000  268.290009  268.899994   
2026-02-03 17:30:00+00:00  269.339996  270.369995  268.755005  268.890015   
2026-02-03 18:30:00+00:00  271.480011  271.875000  268.850006  269.334991   
2026-02-03 19:30:00+00:00  269.209991  271.700012  268.670013  271.470001   
2026-02-03 20:30:00+00:00  269.459991  270.049988  268.839996  269.214996   

Price                        Volume  
Ticker                         AAPL  
Datetime                             
2026-02-02 18:30:00+00:00   7990643  
2026-02-02 19:30:00+00:00   6110168  
2026-02-02 20:30:00+00:00   6031354  
2026-02-03 14:30:00+00:00   9073789  
2026-02-03 15:30:00+00:00   5335029  
2026-02-03 16:30:00+00:00   3977028  
2026-02-03 17:30:00+00:00   3592052  
2026-02-03 18:30:00+00:00  15514895  
2026-02-03 19:30:00+00:00   3448770  
2026-02-03 20:30:00+00:00   3048085

<class 'pandas.DataFrame'>
DatetimeIndex: 26 entries, 2026-01-28 14:30:00+00:00 to 2026-02-03 20:30:00+00:00
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   (Close, AAPL)   26 non-null     float64
 1   (High, AAPL)    26 non-null     float64
 2   (Low, AAPL)     26 non-null     float64
 3   (Open, AAPL)    26 non-null     float64
 4   (Volume, AAPL)  26 non-null     int64  
dtypes: float64(4), int64(1)
memory usage: 1.2 KB


None

Price        Close        High         Low        Open        Volume
Ticker        AAPL        AAPL        AAPL        AAPL          AAPL
count    26.000000   26.000000   26.000000   26.000000  2.600000e+01
mean    261.314423  262.390189  260.249427  261.220193  5.260546e+06
std       6.573289    6.666133    6.448894    6.378315  4.014670e+06
min     254.000000  254.949997  252.179993  254.080002  5.951020e+05
25%     256.110001  256.934998  255.067501  256.077499  3.082354e+06
50%     257.762497  258.777512  256.854996  257.815002  3.784540e+06
75%     268.960007  270.289993  267.669998  268.897499  6.090464e+06
max     271.480011  271.875000  268.850006  271.470001  1.841129e+07

In [3]:
data_history = yf.download(ticker, start="2024-12-01", end="2025-12-31", interval="1d")

display(data_history.shape)

display(data_history.head(10))

display(data_history.tail(10))

display(data_history.info())

display(data_history.describe())

[*********************100%***********************]  1 of 1 completed


(270, 5)

Price            Close        High         Low        Open    Volume
Ticker            AAPL        AAPL        AAPL        AAPL      AAPL
Date                                                                
2024-12-02  238.511261  239.705855  236.092209  236.201715  48137100
2024-12-03  241.557495  241.667000  237.824379  238.730285  38861000
2024-12-04  241.915863  243.010916  240.163793  241.776494  44383900
2024-12-05  241.945724  243.438971  241.039833  242.891459  40033900
2024-12-06  241.746643  243.528592  240.990070  241.816335  36870600
2024-12-09  245.639038  246.126837  240.661550  240.741192  44649200
2024-12-10  246.654449  247.092471  244.235382  245.778407  36914800
2024-12-11  245.380203  249.670795  245.151228  246.843586  45205800
2024-12-12  246.843597  247.620084  244.573849  245.778408  32777500
2024-12-13  247.012817  248.167583  245.131328  246.704216  33155300

Price            Close        High         Low        Open     Volume
Ticker            AAPL        AAPL        AAPL        AAPL       AAPL
Date                                                                 
2025-12-16  274.609985  275.500000  271.790009  272.820007   37648600
2025-12-17  271.839996  276.160004  271.640015  275.010010   50138700
2025-12-18  272.190002  273.630005  266.950012  273.609985   51630700
2025-12-19  273.670013  274.600006  269.899994  272.149994  144632000
2025-12-22  270.970001  273.880005  270.510010  272.859985   36571800
2025-12-23  272.359985  272.500000  269.559998  270.839996   29642000
2025-12-24  273.809998  275.429993  272.200012  272.339996   17910600
2025-12-26  273.399994  275.369995  272.859985  274.160004   21521800
2025-12-29  273.760010  274.359985  272.350006  272.690002   23715200
2025-12-30  273.079987  274.079987  272.279999  272.809998   22139600

<class 'pandas.DataFrame'>
DatetimeIndex: 270 entries, 2024-12-02 to 2025-12-30
Data columns (total 5 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   (Close, AAPL)   270 non-null    float64
 1   (High, AAPL)    270 non-null    float64
 2   (Low, AAPL)     270 non-null    float64
 3   (Open, AAPL)    270 non-null    float64
 4   (Volume, AAPL)  270 non-null    int64  
dtypes: float64(4), int64(1)
memory usage: 12.7 KB


None

Price        Close        High         Low        Open        Volume
Ticker        AAPL        AAPL        AAPL        AAPL          AAPL
count   270.000000  270.000000  270.000000  270.000000  2.700000e+02
mean    232.908428  235.320908  230.303162  232.647364  5.368358e+07
std      25.925164   25.634408   26.116929   25.942791  2.352529e+07
min     171.832428  189.691359  168.633375  171.364011  1.791060e+07
25%     210.701385  212.288889  208.191485  210.127598  4.015680e+07
50%     232.199814  234.355623  229.122896  231.758940  4.728840e+07
75%     253.092838  254.689527  249.379632  252.549434  5.622605e+07
max     286.190002  288.619995  283.299988  286.200012  1.843959e+08

## Como obter sua API key do FRED:

1. **Acesse:** https://fred.stlouisfed.org
2. **Clique em "My Account"** (canto superior direito)
3. **Cadastre-se gratuitamente** (se ainda não tiver conta)
4. **Vá para "Account Settings" → "API Keys"**
5. **Clique em "Request API Key"**
6. **Preencha o formulário** (uso pessoal/educacional é gratuito)

## Depois de obter sua chave:

```python
from fredapi import Fred

# Substitua com SUA chave
fred = Fred(api_key="sua_chave_aqui_32_caracteres")

# Buscar dados de inflação
inflation = fred.get_series("CPIAUCSL")
print(inflation)
```

## Códigos úteis do FRED:
- `CPIAUCSL`: Índice de Preços ao Consumidor (inflação)
- `UNRATE`: Taxa de desemprego
- `GDP`: PIB
- `FEDFUNDS`: Taxa de juros dos Fed Funds

A API key gratuita tem limite de 1.200 requisições por hora, o que é suficiente para a maioria dos usos pessoais e educacionais.

In [4]:
from fredapi import Fred

with open(r"..\..\API_key_FRED.txt", 'r') as f:
    api_key = f.read().strip()

fred = Fred(api_key=api_key)


In [5]:
inflation = fred.get_series("CPIAUCSL")

display(inflation.shape)

display(inflation.head(10))

display(inflation.tail(10))

display(inflation.info())

display(inflation.describe())

(948,)

1947-01-01    21.48
1947-02-01    21.62
1947-03-01    22.00
1947-04-01    22.00
1947-05-01    21.95
1947-06-01    22.08
1947-07-01    22.23
1947-08-01    22.40
1947-09-01    22.84
1947-10-01    22.91
dtype: float64

2025-03-01    319.615
2025-04-01    320.321
2025-05-01    320.580
2025-06-01    321.500
2025-07-01    322.132
2025-08-01    323.364
2025-09-01    324.368
2025-10-01        NaN
2025-11-01    325.031
2025-12-01    326.030
dtype: float64

<class 'pandas.Series'>
DatetimeIndex: 948 entries, 1947-01-01 to 2025-12-01
Series name: None
Non-Null Count  Dtype  
--------------  -----  
947 non-null    float64
dtypes: float64(1)
memory usage: 14.8 KB


None

count    947.000000
mean     123.848383
std       89.214260
min       21.480000
25%       32.800000
50%      109.600000
75%      199.550000
max      326.030000
dtype: float64

In [6]:
unrate = fred.get_series("UNRATE")

display(unrate.shape)

display(unrate.head(10))

display(unrate.tail(10))

display(unrate.info())

display(unrate.describe())

(936,)

1948-01-01    3.4
1948-02-01    3.8
1948-03-01    4.0
1948-04-01    3.9
1948-05-01    3.5
1948-06-01    3.6
1948-07-01    3.6
1948-08-01    3.9
1948-09-01    3.8
1948-10-01    3.7
dtype: float64

2025-03-01    4.2
2025-04-01    4.2
2025-05-01    4.3
2025-06-01    4.1
2025-07-01    4.3
2025-08-01    4.3
2025-09-01    4.4
2025-10-01    NaN
2025-11-01    4.5
2025-12-01    4.4
dtype: float64

<class 'pandas.Series'>
DatetimeIndex: 936 entries, 1948-01-01 to 2025-12-01
Series name: None
Non-Null Count  Dtype  
--------------  -----  
935 non-null    float64
dtypes: float64(1)
memory usage: 14.6 KB


None

count    935.000000
mean       5.666524
std        1.706049
min        2.500000
25%        4.300000
50%        5.500000
75%        6.700000
max       14.800000
dtype: float64

In [7]:
pib = fred.get_series("GDP")

display(pib.shape)

display(pib.head(10))

display(pib.tail(10))

display(pib.info())

display(pib.describe())

(319,)

1946-01-01        NaN
1946-04-01        NaN
1946-07-01        NaN
1946-10-01        NaN
1947-01-01    243.164
1947-04-01    245.968
1947-07-01    249.585
1947-10-01    259.745
1948-01-01    265.742
1948-04-01    272.567
dtype: float64

2023-04-01    27530.055
2023-07-01    28074.846
2023-10-01    28424.722
2024-01-01    28708.161
2024-04-01    29147.044
2024-07-01    29511.664
2024-10-01    29825.182
2025-01-01    30042.113
2025-04-01    30485.729
2025-07-01    31098.027
dtype: float64

<class 'pandas.Series'>
DatetimeIndex: 319 entries, 1946-01-01 to 2025-07-01
Series name: None
Non-Null Count  Dtype  
--------------  -----  
315 non-null    float64
dtypes: float64(1)
memory usage: 5.0 KB


None

count      315.000000
mean      7675.441851
std       8069.026559
min        243.164000
25%        826.470000
50%       4545.340000
75%      13461.682000
max      31098.027000
dtype: float64

In [8]:
fedfunds = fred.get_series("FEDFUNDS")

display(fedfunds.shape)

display(fedfunds.head(10))

display(fedfunds.tail(10))

display(fedfunds.info())

display(fedfunds.describe())

(859,)

1954-07-01    0.80
1954-08-01    1.22
1954-09-01    1.07
1954-10-01    0.85
1954-11-01    0.83
1954-12-01    1.28
1955-01-01    1.39
1955-02-01    1.29
1955-03-01    1.35
1955-04-01    1.43
dtype: float64

2025-04-01    4.33
2025-05-01    4.33
2025-06-01    4.33
2025-07-01    4.33
2025-08-01    4.33
2025-09-01    4.22
2025-10-01    4.09
2025-11-01    3.88
2025-12-01    3.72
2026-01-01    3.64
dtype: float64

<class 'pandas.Series'>
DatetimeIndex: 859 entries, 1954-07-01 to 2026-01-01
Series name: None
Non-Null Count  Dtype  
--------------  -----  
859 non-null    float64
dtypes: float64(1)
memory usage: 13.4 KB


None

count    859.000000
mean       4.602678
std        3.541000
min        0.050000
25%        1.910000
50%        4.320000
75%        6.125000
max       19.100000
dtype: float64

# import feedparser

- entry.title        # título da notícia
- entry.link         # link da matéria
- entry.published    # data de publicação (string)
- entry.summary      # resumo (quando disponível)

In [9]:
import feedparser
rss_url = "https://news.google.com/rss/search?q=AAPL+stock"
feed = feedparser.parse(rss_url)
print(feed.entries[0].keys())


dict_keys(['title', 'title_detail', 'links', 'link', 'id', 'guidislink', 'published', 'published_parsed', 'summary', 'summary_detail', 'source'])


In [10]:
import feedparser

rss_url = "https://news.google.com/rss/search?q=AAPL+stock"
feed = feedparser.parse(rss_url)

for entry in feed.entries[:5]:
    print(entry.title)


Apple Stock Could Surge to $300: Here’s What Needs to Happen - 24/7 Wall St.
AAPL Just Crossed Over a Historically Bullish Trendline - Yahoo Finance
What's Driving The Jump In Apple Stock? - Benzinga
Apple CEO sends blunt message iPhone 18 fans can’t ignore - thestreet.com
AAPL Hold, MSFT Buy Upgrades – Which Tech Giant Wins? - TipRanks
